src/app/store/chat/chat.reducer.ts

```diff
@@ -52,7 +52,10 @@ export const chatFeature = createFeature({
       (state, { message, pendingMessageId }) => ({
         ...state,
         messages: messageAdapter.mapOne(
-          { id: pendingMessageId, map: () => message },
+          {
+            id: pendingMessageId,
+            map: () => ({ ...message, id: pendingMessageId }),
```

Comment:
This is for not to make new ID when we update the existed item.

Reply:
this was actually intended behavior. A temporary id is created so the message is added to the collection to appear before there's a response from the server, but when the response comes back, the server sends the correct ID.

Reply:
No problem with it, you only need to implement one more ID for your case. I mean the UI ID (Scroller's ID), which will persist across the server changes. The Scroller's message is a human-like message remaining the same entity not depending of its contents changes. Status or text (or Server ID) may change, Scroller's ID must not.